crates/sncast/src/main.rs

```diff
@@ -175,19 +172,19 @@ fn main() -> Result<()> {
 async fn run_async_command(
     cli: Cli,
     config: CastConfig,
-    provider: JsonRpcClient<HttpTransport>,
     numbers_format: NumbersFormat,
     output_format: OutputFormat,
 ) -> Result<()> {
-    verify_and_warn_if_incompatible_rpc_version(&provider, &config.url).await?;
-
     let wait_config = WaitForTx {
         wait: cli.wait,
         wait_params: config.wait_params,
     };
 
     match cli.command {
         Commands::Declare(declare) => {
+            let provider = get_provider(&config.url)?;
```

Comment:
```rust
let provider = get_provider(&config.url)?;
verify_and_warn_if_incompatible_rpc_version(&provider, &config.url).await?;
```
You always repeat this sequence; please make this a function. Also, it could be declared in the top scope as lazy value, but it is up to you